Calibration/HcalAlCaRecoProducers/plugins/AlCaHcalIsotrkFilter.cc

```diff
@@ -0,0 +1,183 @@
+        low = true;
+      } else if (itr->p_ > pTrackHigh_) {
+        high = true;
+      } else {
+        inRange = true;
+      }
+    }
+#ifdef EDM_ML_DEBUG
+    edm::LogVerbatim("HcalIsoTrack") << "AlCaHcalIsotrkFilter::Finds " << isotrkCalib->size()
+                                     << " entries in HcalIsoTrkCalibVariables collection with inRange " << inRange
+                                     << " low " << low << " high " << high;
+#endif
+    if (inRange) {
+      accept = true;
+    } else {
+      if (low) {
+        ++nLow_;
+        if (prescaleLow_ <= 1)
+          accept = true;
+        else if (nLow_ % prescaleLow_ == 1)
+          accept = true;
+      }
+      if (high) {
+        ++nHigh_;
+        if (prescaleHigh_ <= 1)
+          accept = true;
+        else if (nHigh_ % prescaleHigh_ == 1)
+          accept = true;
+      }
+    }
+  } else {
+    edm::LogVerbatim("HcalIsoTrack") << "AlCaHcalIsotrkFilter::Cannot find the collection for HcalIsoTrkCalibVariables";
+  }
+
+  // Return the acceptance flag
+  if (accept)
+    ++nGood_;
+#ifdef EDM_ML_DEBUG
+  edm::LogVerbatim("HcalIsoTrack") << "AlCaHcalIsotrkFilter::Accept flag " << accept << " All " << nAll_ << " Good "
+                                   << nGood_ << " Low " << nLow_ << " High " << nHigh_;
+#endif
+  return accept;
+
+}  // AlCaHcalIsotrkFilter::filter
+
+// ------------ method called once each job just after ending the event loop  ------------
+void AlCaHcalIsotrkFilter::endStream() {
+  globalCache()->nAll_ += nAll_;
+  globalCache()->nGood_ += nGood_;
+}
+
+void AlCaHcalIsotrkFilter::globalEndJob(const alcaHcalIsoTrk::Counters* count) {
+  edm::LogVerbatim("HcalIsoTrack") << "Selects " << count->nGood_ << " good events out of " << count->nAll_
+                                   << " total # of events\n";
+}
+
+// ------------ method called when starting to processes a run  ------------
+void AlCaHcalIsotrkFilter::beginRun(edm::Run const& iRun, edm::EventSetup const& iSetup) {
+  edm::LogVerbatim("HcalIsoTrack") << "Run[" << nRun_ << "] " << iRun.run();
+}
+
+// ------------ method called when ending the processing of a run  ------------
+void AlCaHcalIsotrkFilter::endRun(edm::Run const& iRun, edm::EventSetup const&) {
+  edm::LogVerbatim("HcalIsoTrack") << "endRun[" << nRun_ << "] " << iRun.run() << "\n";
+  nRun_++;
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void AlCaHcalIsotrkFilter::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
+  edm::ParameterSetDescription desc;
+  desc.add<double>("momentumLow", 40.0);
+  desc.add<double>("momentumHigh", 60.0);
+  desc.add<int>("prescaleLow", 1);
+  desc.add<int>("prescaleHigh", 1);
+  desc.add<std::string>("isoTrackLabel", "HcalIsoTrack");
```

Comment:
@bsunanda If the `AlCaHcalIsotrkFilter` is expected to read the output of `AlCaHcalIsotrkProducer.cc`, this should be 
```suggestion
  desc.add<std::string>("isoTrackLabel", "alcaHcalIsotrkProducer");
```
Furthermore, please use `edm::InputTag` for Event products
```suggestion
  desc.add<edm::InputTag>("isoTrackLabel", edm::InputTag("alcaHcalIsotrkProducer"));
```

Reply:
I changed the line
desc.add<std::string>("isoTrackLabel", "HcalIsoTrack");
to
desc.add<edm::InputTag>("isoTrackLabel", edm::InputTag("alcaHcalIsotrkProducer"));
But still the filter does not find the collection. But if one removes the filter from the tree one finds the collection in the output

Reply:
I missed the producer used a product label for the products. The correct InputTag is then
```suggestion
  desc.add<edm::InputTag>("isoTrackLabel", edm::InputTag("alcaHcalIsotrkProducer", "HcalIsoTrack"));
```

Note that the product labels are not strictly necessary in the producer since the product types are different (and many times unnecessary product labels lead to confusion).